### PR TITLE
feat(hooks): Provide static getHook method for access to all html-webpack-plugin hooks

### PR DESCRIPTION
**Pull request by jantimon**
Export all hooks static and typed similar to https://github.com/webpack/webpack/pull/7672
cc @ooflorent

### CODE REVIEW

**Comment by jantimon** [comment]
Okay changed it to getHooks

**Comment by emilio-martinez** [comment]
This feels like a great change, in my opinion. The access to hooks feels much cleaner.

File: lib/hooks.js [hunk 1]
```diff
@@ -101,32 +94,18 @@ function getHtmlWebpackPluginHooks (compilation) {
...
-  /** @type {HtmlWebpackPluginHooks} */
-  const hooks = compilation.hooks;
-  hooks.htmlWebpackPluginBeforeHtmlGeneration = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginBeforeHtmlProcessing = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAlterAssetTags = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAfterHtmlProcessing = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAfterEmit = new AsyncSeriesWaterfallHook(['pluginArgs']);
+function createHtmlWebpackPluginHooks () {
+  return {
+    htmlWebpackPluginBeforeHtmlGeneration: new AsyncSeriesWaterfallHook(['pluginArgs']),
```
Comment (sokra) [inline lib/hooks.js]: You should consider removing the `htmlWebpackPlugin` prefix, as it's now already namespace via `getHooks`.

Comment (jantimon) [reply lib/hooks.js]: 👍 I'll take a look

**Pull request merged**